hw3/01-routing.js

```diff
@@ -37,13 +26,87 @@ const server = http.createServer((req, res) => {
   if (req.url === '/') {
     let routeResults = getRoutes();
 
-    res.writeHead(200, { 'Content-Type': 'text/html' });
     res.write(`<h1>Exercise 01</h1>`);
     res.write(`<ul> ${routeResults} </ul>`);
     res.end();
   }
 
   // Add your code here
+
+  // http://localhost:5001/welcome should return a status code 200 with a welcome message of your choice in html format
+  else if (req.method === 'GET' && req.url === '/welcome') {
+    res.writeHead(200, { 'Content-Type': 'text/html' });
+    res.write('<h1>Hello! This is the welcome page</h1>');
+    res.end();
+  }
+
+  // http://localhost:5001/redirect should redirect the request to '/redirected' by using 302 as the status code / the
+  // redirected page should return a redirected message of your choice
+  else if (req.method === 'GET' && req.url === '/redirect') {
+    console.log('redirecting...');
+    res.writeHead(302, { Location: '/redirected' });
+    res.end();
+  }
+
+  // redirected
+  else if (req.method === 'GET' && req.url === '/redirected') {
```

Comment:
I also like how your conditional checks whether it's a get request or not. I think that makes routing much more readable.